src/Parser.js

```diff
@@ -141,15 +150,15 @@ Parser.prototype.parseExpression = function(pos, breakOnInfix, breakOnToken) {
 
             break;
         }
-        if (breakOnInfix && atom.result.type === "infix") {
+        if (breakOnInfix && atom.type === "infix") {
+            // rewind so we can parse the infix atom again
+            this.pos = pos;
+            this.nextToken = lex;
             break;
         }
-        body.push(atom.result);
-        pos = atom.position;
+        body.push(atom);
     }
-    var res = new ParseResult(this.handleInfixNodes(body), pos);
-    res.peek = lex;
-    return res;
+    return this.handleInfixNodes(body);
```

Comment:
Looking at this code again, I realize that I could've done this in a better way.